tracker/view/show.py

```diff
@@ -511,6 +519,32 @@ def show_generated_advisory(advisory_id, raw=False):
         return not_found()
 
     advisory = entries[0][0]
+
+    form = ReviewForm()
+    review = (db.session.query(Review, Advisory, User)
+               .filter(Advisory.id == advisory_id)
+               .filter(User.id == current_user.id)
+               .join(Advisory).join(User)).first()
+    if not form.is_submitted() and review:
+        form.note.data = review[0].note
+    if form.is_submitted():
+        if not form.validate_on_submit():
+            flash("Need to write feedback to disapprove", 'warning')
+            return redirect('/{}/generate'.format(advisory.id))
+        if review:
+            db.session.delete(review[0])
```

Comment:
I'm assuming this is deleting the review that was generated before before committing the new one. I'm not sure this logic is quire parseable due to the `first()` query moifier there and the [0] being I guess the id of the review? I wonder if we could re-structure this for readability.